WorkflowTesting/Sources/Internal/RenderExpectations.swift

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if DEBUG
+
+    @testable import Workflow
+
+    extension RenderTester {
+        internal class AnyExpectedWorkflow {
+            let workflowType: Any.Type
+            let key: String
+            fileprivate init(workflowType: Any.Type, key: String) {
+                self.workflowType = workflowType
+                self.key = key
+            }
+        }
+
+        internal class ExpectedWorkflow<ExpectedWorkflowType: Workflow>: AnyExpectedWorkflow {
+            let rendering: ExpectedWorkflowType.Rendering
+            let output: ExpectedWorkflowType.Output?
+
+            init(key: String, rendering: ExpectedWorkflowType.Rendering, output: ExpectedWorkflowType.Output?) {
```

Comment:
I will need something like in: https://github.com/square/workflow-swift/pull/11/files#diff-d274c3ab6b1b700864fdd0b96d30b731R129

to support custom matching for `Worker`s.

Reply:
Do we? We don’t actually want to reproduce/change the matching behavior since the RenderContext won’t have additional matching logic for Workers. I’d pictured something like this:
```swift
extension RenderTester {
    public func expect<ExpectedWorkerType: Worker>(
        worker expectedWorker: ExpectedWorkerType,
        key: String = "",
        producingOutput output: ExpectedWorkerType.Output? = nil
    ) -> RenderTester<WorkflowType> {
        return self.expectWorkflow(
            type: WorkerWorkflow<ExpectedWorkerType>,
            key: key,
            producingRendering: (),
            producingOutput: output,
            assertions: { actualWorker in
                XCTAssert(expectedWorker.isEquivalent(to: actualWorker))
            }
        )
    }
}
```

(though I _do_ think I should update `assertions` to give the file and line so we can correctly place the error where it should be)

Reply:
Ooh, that's right, fantastic!

A slight change, we'll get the `Workflow` here.

```swift
extension RenderTester {
    public func expect<ExpectedWorkerType: Worker>(
        worker expectedWorker: ExpectedWorkerType,
        key: String = "",
        producingOutput output: ExpectedWorkerType.Output? = nil
    ) -> RenderTester<WorkflowType> {
        return self.expectWorkflow(
            type: WorkerWorkflow<ExpectedWorkerType>,
            key: key,
            producingRendering: (),
            producingOutput: output,
            assertions: { actualWorkflow in
                XCTAssert(actualWorkflow.worker.isEquivalent(to: expectedWorker))
            }
        )
    }
}
```

Reply:
Ah, right, yep!